examples/nyxstone-cli.cpp

```diff
@@ -1,176 +1,234 @@
-    ;
-    // clang-format on
-
-    desc.add(desc_asm);
-    desc.add(desc_disasm);
-
-    po::variables_map varmap;
-    try {
-        po::store(po::parse_command_line(argc, argv, desc), varmap);
-    } catch (const std::exception& e) {
-        std::cerr << "Error occured: " << e.what() << "\n";
-        return 2;
+    auto input_size = hex_string.size();
+    if (input_size % 2 != 0) {
+        return {};
     }
-    po::notify(varmap);
 
-    if (varmap.empty() || varmap.count("help") != 0) {
-        std::cout << desc << "\n";
-        return 0;
+    std::vector<uint8_t> result;
+    result.reserve(input_size / 2);
+
+    try {
+        for (size_t i = 0; i < input_size; i += 2) {
+            result.emplace_back(std::stoul(hex_string.substr(i, 2), nullptr, 16));
+        }
+    } catch (...) {
+        return {};
     }
 
-    const bool has_assemble = varmap.count("assemble") > 0;
-    const bool has_disassemble = varmap.count("disassemble") > 0;
+    return result;
+}
 
-    if (!has_assemble && !has_disassemble) {
-        std::cout << "Nothing to do\n";
-        std::cout << desc << "\n";
-        return 1;
+constexpr auto USAGE = R"(Usage: nyxstone [-t=<triple>] [-p=<pc>] [-d] <input>
+
+Examples:
+  # Assemble an instruction with the default architecture ('x86_64').
+  nyxstone 'push eax'
+
+  # Disassemble the bytes 'ffc300d1' as AArch64 code.
+  nyxstone -t aarch64 -d ffc300d1
+
+Options:
+  -t, --triple=<triple>      LLVM target triple or alias, e.g. 'aarch64'
+  -c, --cpu=<cpu>            LLVM CPU specifier, e.g. 'cortex-a53'
+  -f, --features=<list>      LLVM architecture/CPU feature list, e.g. '+mte,-neon'
+  -p, --address=<pc>         Initial address to assemble/disassemble relative to
+  -l, --labels=<list>        Label-to-address mappings (used when assembling only)
+  -d, --disassemble          Treat <input> as bytes to disassemble instead of assembly
+  -h, --help                 Show this help and usage message
+
+Notes:
+  The '--target' parameter also supports aliases for common target triples:
+
+     'x86_32' -> 'i686-linux-gnu'
+     'x86_64' -> 'x86_64-linux-gnu'
+     'armv6m' -> 'armv6m-none-eabi'
+     'armv7m' -> 'armv7m-none-eabi'
+     'armv8m' -> 'armv8m.main-none-eabi'
+    'aarch64' -> 'aarch64-linux-gnueabihf'
+
+  The CPUs for a target can be found with 'llc -mtriple=<triple> -mcpu=help'.
+  The features for a target can be found with 'llc -mtriple=<triple> -mattr=help'.
+)";
+
+/// Parsed program options.
+struct Options {
+    bool is_valid = false;
+    std::string error;
+
+    std::string triple = "x86_64-linux-gnu";
+    std::string cpu;
+    std::string features;
+    uint64_t address = 0;
+    std::vector<Nyxstone::LabelDefinition> labels;
+    bool disassemble = false;
+    bool show_help = false;
+
+    std::string input;
+
+    Options(int argc, char const** argv);
+};
+
+Options::Options(int argc, char const** argv)
```

Comment:
I think it would be more in-line with existing code to have some fallible function `parse` (or similar) instead, which returns a `tl::expected<Options, std::string>`. Should also remove the need for the members `valid` and `error`.